AxonDeepSeg/testing/segmentation_scoring.py

```diff
@@ -28,6 +29,46 @@ def score_analysis(img, groundtruth, prediction, visualization=False, min_area=2
     :param min_area: minimal area of the predicted axon to count.
     :return: [sensitivity, precision, diffusion]
     """
+    if visualization:
+        # Define helper functions for plotting
+        def _create_figure(result_img, title):
+            """
+            Full OO syntax for plotting a segmentation and centroids 
+            over the base image.
+            Fuction made for internal usage because it depends on the closure 
+            of local variables.
+            NB: In the future, `score_analysis` should be refactored 
+            as a class.
+            :param result_img: result of the segmentation
+            :param title: str to add to the title of the figure
+            :return: matplotlib.figure.Figure
+            """
+            fig = Figure()
+            FigureCanvas(fig)
+            ax = fig.subplots()
+            # Plot base image and segmentation result
+            ax.imshow(img, cmap="gray")
+            ax.imshow(result_img, alpha=0.7)
+            # Add centroids
+            ax.scatter(centroids_T[:, 1], centroids_T[:, 0], color="g")
+            ax.scatter(centroids_F[:, 1], centroids_F[:, 0], color="r")
+            ax.scatter(notDetected[:, 1], notDetected[:, 0], color="y")
+            # Add title
+            ax.set_title(
+                title
+                + ", Sensitivity : %s , Precision : %s " % (sensitivity, precision)
+            )
+            return fig
+
+        def _save_prediction_and_ground_truth_images(prediction_img, groundtruth_img):
+            """
+            Plot and save pred and GT images. 
+            """
+            fig1 = _create_figure(prediction_img, "Prediction")
+            fig2 = _create_figure(groundtruth_img, "Ground Truth")
+
+            fig1.savefig("./prediction.png")
```

Comment:
@maf88 These two files (prediction.png and ground_truth.png) are new, so they need to be cleaned up (i.e. deleted) in the teardown of the test function for this file (i.e. [here](https://github.com/neuropoly/axondeepseg/blob/b85496e7c4036c66dd03faef6c290ba5a64306a6/test/testing/test_segmentation_scoring.py#L38-L39). Otherwise, they're left over after a test is run, and could accidentally be committed (like I almost did). See [here](https://github.com/neuropoly/axondeepseg/blob/b85496e7c4036c66dd03faef6c290ba5a64306a6/test/test_segment.py#L41-L71) for how I've handled cleaning test-generated files in another test file.